modules/cdp-engine/src/lib.rs

```diff
@@ -777,11 +751,10 @@ impl<T: Config> Pallet<T> {
 		Self::collateral_params(currency_id).required_collateral_ratio
 	}
 
-	pub fn get_interest_rate_per_sec(currency_id: CurrencyId) -> Rate {
+	pub fn get_interest_rate_per_sec(currency_id: CurrencyId) -> Result<Rate, DispatchError> {
 		Self::collateral_params(currency_id)
 			.interest_rate_per_sec
-			.unwrap_or_default()
-			.saturating_add(Self::global_interest_rate_per_sec())
+			.ok_or_else(|| Error::<T>::InvalidCollateralType.into())
```

Comment:
```suggestion
			.ok_or(Error::<T>::InvalidCollateralType.into())
```

Reply:
i think clippy check fails with ok_or

Reply:
hmm that's weird, but doesn't matter it is the same thing